src/platform/Zephyr/ZephyrConfig.cpp

```diff
@@ -229,7 +255,9 @@ CHIP_ERROR ZephyrConfig::WriteConfigValue(Key key, uint64_t val)
 
 CHIP_ERROR ZephyrConfig::WriteConfigValueStr(Key key, const char * str)
 {
-    return WriteConfigValueStr(key, str, str ? strlen(str) : 0);
+    /* The max value size is smaller than the sector size by several ATE size, but the API to get the exact size is not available.
+     * The backend implementation validates the detailed size, so we don't need to do that here. */
+    return WriteConfigValueStr(key, str, str ? strnlen(str, sSectorSize) : 0);
```

Comment:
Honestly, this code doesn't really protect from out of bounds reads as the sector size will most of the time be an order of magnitude bigger than the value provided. The API contract is that `str` must be null-terminated and we have no other choice than to trust it is.

Reply:
I see, thanks for the input

Reply:
I will put a comment there, as we are going to keep getting this usage as potentially error-prone from the coverity. This way we will remember it's safe to ignore it.

Reply:
It seems you could just suppress the warning using coverity annotation: https://doclazy.wordpress.com/2011/07/14/coverity-suppressing-false-positives-with-cod/